src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator.php

```diff
@@ -265,7 +265,7 @@ private function generateSecurityMetadata(Entity $entity)
      */
     private function generateOidcClient(Entity $entity)
     {
-        $metadata['clientId'] = $entity->getEntityId();
+        $metadata['clientId'] = str_replace('://', '@//', $entity->getEntityId());
```

Comment:
Perhaps you could use `ViewObject/Entity::getEntityId` here?

Reply:
No, that is not the responsibility of a viewobject. The replacement is only for Manage to understand the clientId so I find this a much better place.

Reply:
I now see that the entity in this method is a Domain entity and not a View object. I agree with you it would be a hassle to convert that domain entity to a view object just to get the client id.

The best solution might be to introduce a client id conversion helper class that encapsulates the str_replace logic. But lets not go down that route now. This suffices for now!